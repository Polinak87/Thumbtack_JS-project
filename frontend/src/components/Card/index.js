import React from 'react';
import Column from '../Columns/Column';

export default function Card(props) {
  const {
    header,
    button,
    children,
  } = props;

  return (
    <Column>
      <div className="card">
        {header}
        {children}
        {button}
      </div>
    </Column>
  );
}
