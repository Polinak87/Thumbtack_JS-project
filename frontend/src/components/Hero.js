import React from 'react';
import classNames from 'classnames';

export const aquamarine = 'is-primary';
export const blue = 'is-info';

export default function Hero({ className, text }) {
  const classes = () => {
    return classNames('hero', className);
  };

  return (
    <section className={classes()}>
      <div className="hero-body">
        <div className="container">
          <h1 className="title">{text}</h1>
        </div>
      </div>
    </section>
  );
}
