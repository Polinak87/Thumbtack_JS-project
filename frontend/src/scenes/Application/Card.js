import React from 'react';
import CardFooterItem from './CardFooterItem';
import CardButton from './CardButton';
import Header from '../../components/Card/Header'

export default function Card(props) {
  const {
    titleRight,
    titleLeft,
    application,
    applicationType,
    onClickComplete,
    onClickReject,
    onClickCancel,
  } = props;

  const { id, ThingDesired, ThingOffered, status } = application;

  return (
    <div className="card">
      <Header>
        <h6 className="title">Status of application: {status}</h6>
      </Header>
      <footer className="card-footer">
        <CardFooterItem
          title={titleLeft}
          userThing={ThingDesired}
        />
        <CardFooterItem
          title={titleRight}
          userThing={ThingOffered}
        />
      </footer>
      <CardButton
        status={status}
        applicationType={applicationType}
        id={id}
        onClickComplete={onClickComplete}
        onClickReject={onClickReject}
        onClickCancel={onClickCancel}
      />
    </div>
  );
}
