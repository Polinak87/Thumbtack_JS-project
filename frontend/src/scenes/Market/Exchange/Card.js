import React from 'react';
import ThingInfo from '../../../components/ThingInfo';
import Button, { green, large } from '../../../components/Button';

export default class Card extends React.Component {
  constructor(props) {
    super(props);
  }

  render() {
    const {
      image,
      id,
      name,
      description,
      categoryName,
      onMarket,
      onMarketAt,
      onClick,
    } = this.props;

    return (
      <div className="card">
        <ThingInfo
          image={image}
          id={id}
          name={name}
          description={description}
          categoryName={categoryName}
          onMarket={onMarket}
          onMarketAt={onMarketAt}
        />
        <Button className={`${large} ${green}`} id={id} onClick={onClick}>
          Choose
        </Button>
      </div>
    );
  }
}
