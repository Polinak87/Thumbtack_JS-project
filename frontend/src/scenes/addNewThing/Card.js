import React from 'react';
import ThingInfo from '../../components/ThingInfo';
import Button from '../../components/Button'

export default class Card extends React.Component {
  constructor(props) {
    super(props);
  }

  render() {
    const { id, name, description, categoryName, onClick } = this.props;

    return (
      <div className="card">
        <ThingInfo
          id={id}
          name={name}
          description={description}
          categoryName={categoryName}/>
          <Button type='Add to inventory' id={id} onClick={onClick}/>
      </div >
    );
  }
}