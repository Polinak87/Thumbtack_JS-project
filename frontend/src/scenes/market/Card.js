import React from 'react';
import { Link } from 'react-router-dom';
import { connect } from 'react-redux';
import ThingInfo from '../../components/ThingInfo';
import Button from '../../components/Button';
import { addFiltrationByUser } from '../../store/actions/filtration';

class Card extends React.Component {
  constructor(props) {
    super(props);
    this.onClick = this.onClick.bind(this);
  }

  onClick(id) {
    const { addFiltrationByUser } = this.props;
    addFiltrationByUser({id});
  }

  render() {
    const { image, id, name, description, categoryName, onMarket, onMarketAt, user, userId, currentUserId, onClick } = this.props;
    const { firstName, lastName } = user;
    const button = () => {
      if (currentUserId === userId) {
        return;
      } return <Button type='Exchange' id={id} userId={userId} onClick={onClick} />
    }

    return (
      <div className="card">
        <div className="card-header">
          <Link to="/marketthingsfilteredbyuser" className="card-header-title has-text-grey is-centered is-italic is-size-3" onClick={() => this.onClick(user.id)}>
            By {firstName} {lastName}
          </Link>
        </div>
        <ThingInfo
          image={image}
          id={id}
          name={name}
          description={description}
          categoryName={categoryName}
          onMarket={onMarket}
          onMarketAt={onMarketAt} />
        <>
          {button()}
        </>
      </div >
    );
  }
}

const mapDispatchToProps = dispatch => ({
  addFiltrationByUser: (id) => dispatch(addFiltrationByUser(id)),
  dispatch,
});

export default connect(null, mapDispatchToProps)(Card);