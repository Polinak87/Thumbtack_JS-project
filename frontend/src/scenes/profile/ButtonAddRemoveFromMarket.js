import React from 'react';
import axios from 'axios';

export default class ButtonAddRemoveFromMarket extends React.Component {
  constructor(props) {
    super(props);
    this.handleClickAdd = this.handleClickAdd.bind(this);
    this.handleClickRemove = this.handleClickRemove.bind(this);
  }

  handleClickAdd() {
    const id = this.props.id;
    axios.post('/api/addthingtomarket', { id })
    .then((response) => {
      this.props.updateData(id, response.data.onMarket, response.data.onMarketAt);
    });
  }

  handleClickRemove() {
    event.preventDefault();
    const id = this.props.id;
    axios.post('/api/removethingfrommarket', { id })
      .then((response) => {
        this.props.updateData(id, response.data.onMarket, response.data.onMarketAt);
      });
  }

  render() {
    const { onMarket } = this.props;
    let button;
    if (onMarket) {
      button = <button className="button is-block is-danger is-large is-fullwidth" onClick={this.handleClickRemove} >Remove from market</button>;
    } else {
      button = <button className="button is-block is-success is-large is-fullwidth" onClick={this.handleClickAdd} >Add to market</button>;
    }

    return (
      <>
        {button}
      </>
    );
  }
}