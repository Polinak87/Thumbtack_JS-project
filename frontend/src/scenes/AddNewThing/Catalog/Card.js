import React from 'react';
import ThingInfo from '../../../components/ThingInfo';
import Button from '../../../components/Button';
import { green, large } from '../../../components/Button';

export default class Card extends React.Component {
  render() {
    const { image, id, name, description, categoryName, onClick } = this.props;

    return (
      <div className="card">
        <ThingInfo
          image={image}
          id={id}
          name={name}
          description={description}
          categoryName={categoryName}/>
          <Button className={large + " " + green} value='Add to inventory' id={id} onClick={onClick}/>
      </div >
    );
  }
}