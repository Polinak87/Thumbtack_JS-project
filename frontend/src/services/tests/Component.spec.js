import React from 'react';
import { render, fireEvent, cleanup } from 'react-testing-library';
import Hero from '../../components/hero';
import ButtonAddRemoveFromMarket from  '../../scenes/profile/buttonAddRemoveFromMarket';
import Card from '../../scenes/profile/card';

afterEach(cleanup);

it('should has hero name', () => {
  const heroText = 'Your inventory';
  const { getByText } = render(<Hero heroText={heroText}/>);
  expect(getByText(heroText).textContent).toBe(heroText);
});

it('clicks button', () => {
  const onMarket = false;
  const { getByText } = render(<ButtonAddRemoveFromMarket onMarket={onMarket} />);
  fireEvent.click(getByText('Add to market'));
});

it('snapshort of button', () => {
  const onMarket = true;
  const { container } = render(<ButtonAddRemoveFromMarket onMarket={onMarket} />);
  expect(container.firstChild).toMatchSnapshot();
});

it('snapshort of card', () => {
  const id=1;
  const name='summer dress';
  const description='light';
  const categoryName='dresses';
  const onMarket=true;
  const onMarketAt='2019-05-15T05:21:56.000Z';
  const { container } = render(<Card
                                id={id}
                                name={name}
                                description={description}
                                categoryName={categoryName}
                                onMarket={onMarket}
                                onMarketAt={onMarketAt}/>);
  expect(container.firstChild).toMatchSnapshot();
});

