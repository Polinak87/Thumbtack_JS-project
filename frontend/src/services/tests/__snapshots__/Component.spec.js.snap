// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`snapshort of button 1`] = `
<button
  class="button is-block is-danger is-large is-fullwidth"
>
  Remove from market
</button>
`;

exports[`snapshort of card 1`] = `
<div
  class="card"
>
  <div
    class="card-image"
  >
    <figure
      class="image is-4by3"
    >
      <img
        alt="Placeholder image"
        src="https://bulma.io/images/placeholders/1280x960.png"
      />
    </figure>
  </div>
  <div
    class="card-content"
  >
    id: 
    1
    <br />
    Name: 
    summer dress
    <br />
    Category: 
    dresses
    <br />
    Description: 
    light
    <br />
    <time
      datetime="2016-1-1"
    >
      On market from: 
      2019-05-15T05:21:56.000Z
    </time>
  </div>
  <button
    class="button is-block is-danger is-large is-fullwidth"
  >
    Remove from market
  </button>
</div>
`;
